### PR TITLE
Initial k3s docs

### PR DESCRIPTION
**Pull request by erikwilson**

### CODE REVIEW

**Comment by davidnuzik** [comment]
I am reviewing currently.

File: content/k3s/latest/en/running/_index.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+If you installed k3s with the help of `install.sh` script an uninstall script is generated during installation, which will be created on your server node at `/usr/local/bin/k3s-uninstall.sh` (or as `k3s-agent-uninstall.sh`).
+
+Hyperkube
+---------
+
+k3s is bundled in a nice wrapper to remove the majority of the headache of running k8s. If
+you don't want that wrapper and just want a smaller k8s distro, the releases includes
+the `hyperkube` binary you can use.  It's then up to you to know how to use `hyperkube`. If
+you want individual binaries you will need to compile them yourself from source
```
Comment (davidnuzik) [inline content/k3s/latest/en/running/_index.md]: Needs period at the end of last sentence here.

File: content/k3s/latest/en/running/_index.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+- download **k3s** to **/usr/local/bin/k3s**
+- create an openrc file in **/etc/init.d**
+
+Running in Docker (and docker-compose)
+-----------------
+
+[k3d](https://github.com/rancher/k3d) is a utility designed to easily run k3s in Docker. It can be installed via the [brew](https://brew.sh/) utility for MacOS.
+
+`rancher/k3s` images are also available to run k3s server and agent from Docker.  A `docker-compose.yml` is in the root of the k3s repo that
+serves as an example of how to run k3s from Docker.  To run from `docker-compose` from this repo run
```
Comment (davidnuzik) [inline content/k3s/latest/en/running/_index.md]: Add colon after "run"

File: content/k3s/latest/en/running/_index.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+EOF
+```
+
+Then update **/etc/update-extlinux.conf** by adding:
+
+```
+default_kernel_opts="...  cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory"
+```
+
+Than update the config and reboot
```
Comment (davidnuzik) [inline content/k3s/latest/en/running/_index.md]: Add colon after "reboot"

File: content/k3s/latest/en/installation/_index.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+weight: 2
+---
+
+This section contains information on flags and environment variables used for starting a k3s cluster.
+
+Install Script
+--------------
+
+The install script will attempt to download the latest release, to specify a specific
+version for download we can use the `INSTALL_K3S_VERSION` environment variable, eg:
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/_index.md]: let's avoid "eg" / "e.g." where possible. I think that ", such as in this example:" is okay

File: content/k3s/latest/en/installation/_index.md [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=vX.Y.Z-rc1 sh -
+```
+
+To install just the server without an agent we can add a `INSTALL_K3S_EXEC`
+environment variable to the command:
+```sh
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--disable-agent" sh -
+```
+
+The installer can also be run without performing downloads by setting `INSTALL_K3S_SKIP_DOWNLOAD=true`, eg:
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/_index.md]: let's avoid "eg" / "e.g." where possible. I think that ", such as in this example:" is okay

File: content/k3s/latest/en/quick-start/_index.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+Install Script
+--------------
+The k3s `install.sh` script provides a convenient way for installing to systemd or openrc,
+to install k3s as a service just run:
+```bash
+curl -sfL https://get.k3s.io | sh -
+```
+
+A kubeconfig file is written to `/etc/rancher/k3s/k3s.yaml` and the service is automatically started or restarted.
+The install script will install k3s and additional utilities, such as `kubectl`, `crictl`, `k3s-killall.sh`, and `k3s-uninstall.sh`, eg:
```
Comment (davidnuzik) [inline content/k3s/latest/en/quick-start/_index.md]: let's avoid "eg" / "e.g." where possible. I think that ", such as in this example:" is okay
or ", example:"

File: content/k3s/latest/en/quick-start/_index.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+```bash
+sudo kubectl get nodes
+```
+
+`K3S_TOKEN` is created at `/var/lib/rancher/k3s/server/node-token` on your server.
+To install on worker nodes we should pass `K3S_URL` along with
+`K3S_TOKEN` or `K3S_CLUSTER_SECRET` environment variables, eg:
```
Comment (davidnuzik) [inline content/k3s/latest/en/quick-start/_index.md]: let's avoid "eg" / "e.g." where possible. I think that ", such as in this example:" is okay
or ", example:"

File: content/k3s/latest/en/running/_index.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+            -d --tmpfs /run \
+            --tmpfs /var/run \
+            -e K3S_URL=https://k3s.example.com:6443 \
+            -e K3S_TOKEN=XXX \
+            --privileged rancher/k3s:vX.Y.Z
+
+Air-Gap Support
+---------------
+
+k3s supports pre-loading of containerd images by placing them in the `images` directory for the agent before starting, eg:
```
Comment (davidnuzik) [inline content/k3s/latest/en/running/_index.md]: let's avoid "eg" / "e.g." where possible. I think that ", such as in this example:" is okay
or ", example:"

File: content/k3s/latest/en/running/_index.md [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+sudo ip -c address add 192.168.123.123/24 dev eno1
+sudo ip route add default via 192.168.123.1
+```
+
+k3s additionally provides a `--resolv-conf` flag for kubelets, which may help with configuring DNS in air-gap networks.
+
+Upgrades
+--------
+
+To upgrade k3s from an older version you can re-run the installation script using the same flags, eg:
```
Comment (davidnuzik) [inline content/k3s/latest/en/running/_index.md]: let's avoid "eg" / "e.g." where possible. I think that ", such as in this example:" is okay
or ", example:"

File: content/k3s/latest/en/installation/_index.md [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+
+    (agent) Customized flag for kubelet process
+
+* `--kube-proxy-arg` _value_
+
+    (agent) Customized flag for kube-proxy process
+
+* `--node-label` _value_
+
+    (agent) Registring kubelet with set of labels
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/_index.md]: sp? "Registring" > Registering

File: content/k3s/latest/en/installation/_index.md [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+
+* `--node-taint` _value_
+
+    (agent) Registring kubelet with set of taints
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/_index.md]: sp? "Registring" > Registering

File: content/k3s/latest/en/installation/_index.md [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+
+    (agent) Customized flag for kubelet process
+
+* `--kube-proxy-arg` _value_
+
+    (agent) Customized flag for kube-proxy process
+
+* `--node-label` _value_
+
+    (agent) Registring kubelet with set of labels
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/_index.md]: sp? "Registring" > Registering

File: content/k3s/latest/en/installation/_index.md [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+
+* `--node-taint` _value_
+
+    (agent) Registring kubelet with set of taints
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/_index.md]: sp? "Registring" > Registering

File: content/k3s/latest/en/configuration/_index.md [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+
+k3s includes and defaults to containerd. Why? Because it's just plain better. If you want to
+run with Docker first stop and think, "Really? Do I really want more headache?" If still
+yes then you just need to run the agent with the `--docker` flag
+
+     k3s agent -s ${SERVER_URL} -t ${NODE_TOKEN} --docker &
+
+k3s will generate config.toml for containerd in `/var/lib/rancher/k3s/agent/etc/containerd/config.toml`, for advanced customization for this file you can create another file called `config.toml.tmpl` in the same directory and it will be used instead.
+
+The `config.toml.tmpl` will be treated as a Golang template file, and the `config.Node` structure is being passed to the template,the following is an example on how to use the structure to customize the configuration file https://github.com/rancher/k3s/blob/master/pkg/agent/templates/templates.go#L16-L32
```
Comment (davidnuzik) [inline content/k3s/latest/en/configuration/_index.md]: "template,the" needs space after comma.

File: content/k3s/latest/en/configuration/_index.md [hunk 15]
```diff
@@ -0,0 +1,291 @@
...
+
+Rootless
+--------
+
+_**WARNING**:_ Some advanced magic, user beware
+
+Initial rootless support has been added but there are a series of significant usability issues surrounding it.
+We are releasing the initial support for those interested in rootless and hopefully some people can help to
+improve the usability.  First ensure you have proper setup and support for user namespaces.  Refer to the
+[requirements section](https://github.com/rootless-containers/rootlesskit#setup) in rootlesskit for instructions.
```
Comment (davidnuzik) [inline content/k3s/latest/en/configuration/_index.md]: "rootlesskit" > rootless kit
I think

File: content/k3s/latest/en/configuration/_index.md [hunk 16]
```diff
@@ -0,0 +1,291 @@
...
+    The issue is tracked in https://github.com/rootless-containers/rootlesskit/issues/65
+
+* **Cgroups**
+
+    Cgroups are not supported
+
+**Running w/ Rootless**:
+
+Just add `--rootless` flag to either server or agent.  So run `k3s server --rootless` and then look for the message
+`Wrote kubeconfig [SOME PATH]` for where your kubeconfig to access you cluster is.  Becareful, if you use `-o` to write
```
Comment (davidnuzik) [inline content/k3s/latest/en/configuration/_index.md]: "Becareful" > Be careful

**Comment by davidnuzik** [comment]
Overall review:
- [ ] Does not show up on main / page as a "card" (this is what we want for now).
- [ ] Does show up as drop-down option at top-left in nav panel

Thoughts:
- Do we wish to show the Road Map section?

@deniseschannon are you interested in reviewing as well?

**Comment by deniseschannon** [comment]
Can you share a screenshot of what the card looks like? Since we have 4 products, i'd like to see what it looks like. Since k3s has it's own site, I was thinking docs from k3s website would link directly from that site to the docs site. 

k3s would automatically show up int he drop down of available docs, but I didnt' feel like it needed to be in the landing page of docs from the main rancher site.

**Comment by erikwilson** [comment]
I feel like if I go to docs.rancher.com looking for docs, and then go to k3s.io to find them, them back to rancher docs for the actual docs, that is a worse experience than having some blank space on a page. The docs site should have a primary goal of helping people to find information, and a secondary goal of looking good. 

Here is a screenshot with the blank space of an odd number of cards:
Screen Shot 2019-06-27 at 12 32 46 PM

Here is a screenshot with what I call the 'odd cow' to help fill in that space:
Screen Shot 2019-06-27 at 12 33 03 PM

**Comment by davidnuzik** [comment]
Assuming we still want the card, can we make the cow look less odd? Closer to the graphical logo (eyeless? or does that look odd?), want to share the text?

**Comment by deniseschannon** [comment]
@lvuch maybe you can help make this look better? :) cc: @davidnuzik @erikwilson

**Comment by erikwilson** [comment]
Made the cow a little less odd and a little bit more blue:
Screen Shot 2019-06-27 at 1 33 31 PM

And a commit for the cow, any help appreciated @lvuch  :)

**Comment by davidnuzik** [comment]
Also, just a thought, any harm in re-using rancher logo svg?
https://rancher.com/imgs/rancher-logo-only-color.svg
cc: @lvuch

**Comment by lvuch** [comment]
@erikwilson @davidnuzik How about something like this? I have the changes ready to push if this works.
5grid-centered

**Comment by erikwilson** [comment]
I think that looks great, thanks @lvuch ! 
@deniseschannon looks good to move forward?

**Comment by davidnuzik** [comment]
I agree I think this looks good. Plus at some point we may add Rio in as well.
Thanks @lvuch

**Comment by davidnuzik** [comment]
@erikwilson @lvuch we have an approval from @deniseschannon 
Let's get your changes in please @lvuch 
Once in, I'd like to clone and test for one final review.

Once I have done final review, we may need to touch base with Vince to get this merged in and updated on the website.

**Comment by davidnuzik** [comment]
Reference:
https://github.com/rancher/docs/pull/1569
https://github.com/rancherlabs/website-theme/pull/72

We will use these to actually merge in to the website.

**Comment by davidnuzik** [comment]
Closing this PR in favor of https://github.com/rancher/docs/pull/1569 which has some additional styling changes.

**Pull request closed without merging**